### PR TITLE
Source Mixpanel: added HttpAvailabilityStrategy and undeclared fields for export and annotations streams

### LINKED ISSUE #22262: Source source-mixpanel: implement AvailabilityStrategy
## What
We want GA connectors to make use of an AvailabilityStrategy in their streams to be able to perform error handling on the
stream level. The base use case supported by the `HttpAvailabilityStrategy` is for stream-based permissions,
in order to be able to skip a stream that the connector no longer has access to (because of e.g. scope
changes on an Oauth user config). An `AvailabilityStrategy` can be used to address any error that may occur
on a stream level, to avoid hitting errors, or having to except the same error for every record in a stream,
instead of skipping that stream altogether.

## How
There are a few different ways to go about this, usually depending on how much custom error handling we already
do in a connector, but also due to how the connector works.
- If we have no custom error handling: Inherit the default HttpAvailabilityStrategy. This is as simple as removing the override setting that currently sets
  the AvailabilityStrategy for the streams to `None`. This is useful if we don't currently do much custom error handling
  in the connector.
- If we have custom error handling for 403: Forbidden errors: Do the above to inherit the HttpAvailabilityStrategy,
  removing the custom error handling, but continuing any testing we have that 403s are handled gracefully by the connector
- If we have custom error handling for lots of per-stream errors: Extend the HttpAvailabilityStrategy. You can subclass
  the HttpAvailabilityStrategy to change the `reasons_for_unavailable_status_codes` method to except more error codes and
  provide messages for those error codes.
- If there is a smarter way to determine availability than trying to read the first record and excepting errors: Inherit
  the AvailabilityStrategy base class. For example, if there are clearly defined scopes required to access each stream in
  a given source, you might instead determine whether a stream is available by checking the scopes on the authenticator to
  see if it has the scopes required by the stream.

Note that the AvailabilityStrategy is defined on each stream. For most cases, it is easiest to set it in your stream's
base stream class, where the override is, but for some use cases, different streams might have different availability strategies -
and that's totally fine.

### LINKED ISSUE #24490: Source Mixpanel: Add undeclared columns to spec
## What
In this [PR](https://github.com/airbytehq/airbyte/pull/23985), `BasicReadTest.test_read` test in the Connector Acceptance Tests was updated to fail if the connector produces
stream records which contain columns that haven't been declared in the spec.

Mixpanel currently fails the updated test. Its `acceptance-test-config.yml` was edited with the
`fail_on_extra_columns: false` parameter in order to avoid this change from making the connector fail CAT.

We want to add the undeclared columns to the spec.

**How this will help**:
- Column selection is currently blocked for connectors that don't declare all columns that they pass. This
  is because turning on column selection will stop these undeclared columns from being sent, when they were
  previously being sent. Doing this allows us to enable column selection for Mixpanel!
- Users will better understand the shape of the data they'll receive, since the columns will match the spec.

## How
The following descriptions of streams that pass undeclared columns come from results of the failed connector acceptance test:

```
columns.txt:./source-mixpanel.txt-80-The annotations stream has the following schema errors:
columns.txt:./source-mixpanel.txt:81:Additional properties are not allowed ('user' was unexpected)
columns.txt:./source-mixpanel.txt-100-The export stream has the following schema errors:
columns.txt:./source-mixpanel.txt:101:Additional properties are not allowed ('Referred by', 'import', 'URL', 'insert_id', 'mp_api_timestamp_ms', 'mp_api_endpoint', 'mp_processing_time_ms' were unexpected)

```

1. Add the missing properties indicated by the `Additional properties are not allowed ('<column>', 'column' were unexpected)` logs to the connector's spec.
2. Remove `fail_on_extra_columns: false` from the `acceptance-test-config.yml` file.
3. Commit changes to spec and `acceptance-test-config.yml` and open a PR.
4. Run tests on the connector, either automatically via CI or manually via the `/test` command
5. Profit!

Definition of done: Mixpanel passes CAT without declaring `fail_on_extra_columns: false`. If the
API starts sending over extra columns in the future, we will catch and fix them as part of the #connector-health
movement.

### PR DESCRIPTION
**Pull request by darynaishchenko**
- resolved https://github.com/airbytehq/airbyte/issues/24490 (set HttpAvailabilityStrategy)
- resolved https://github.com/airbytehq/airbyte/issues/22262 (added fields to schema)
Also was updated backoff: set exponential backoff as [docs](https://developer.mixpanel.com/reference/import-events#rate-limits) recommends it. 60s in backoff calculation as delay were set on testing result. 1 request per minute were removed because API allows to send more request per minute, so we can decrease time of work of connector. If we receive 429 error backoff will handle it.

### CODE REVIEW

File: airbyte-integrations/connectors/source-mixpanel/source_mixpanel/streams/base.py [hunk 1]
```diff
@@ -58,7 +57,7 @@ def __init__(
 
     @property
     def availability_strategy(self) -> Optional["AvailabilityStrategy"]:
-        return None
+        return HttpAvailabilityStrategy()
```
Comment (artem1205) [inline airbyte-integrations/connectors/source-mixpanel/source_mixpanel/streams/base.py:60]: you can remove this method from here

File: airbyte-integrations/connectors/source-mixpanel/source_mixpanel/schemas/annotations.json [hunk 2]
```diff
@@ -14,6 +14,20 @@
...
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": ["null", "integer"]
+        },
+        "first_name": {
+          "type" : "string"
+        },
+        "last_name": {
+          "type" : "string"
```
Comment (artem1205) [inline airbyte-integrations/connectors/source-mixpanel/source_mixpanel/schemas/annotations.json:28]: please add `null` in type

**Review by artem1205 — APPROVED** [review]

**Review by lazebnyi — APPROVED** [review]
Nice!

**Review by bnchrch — APPROVED** [review]
Looks good.

Before you merge would you
1. Please make sure you update the version in both the source_definition.yaml and metadata.yaml file
2. Run processResources

**Pull request merged**